cmd/kube-dns/app/server.go

```diff
@@ -172,6 +171,28 @@ func validateHostAndPort(hostAndPort string) error {
 	return nil
 }
 
+// validateNameServer validates a configured name server and
+// appends a ":53" if a :port is not included. IPv6 addresses are also
+// enclosed in square brackets if not already so enclosed.
+func validateNameServer(nameServer string) (string, error) {
+	if ip := net.ParseIP(nameServer); ip != nil {
```

Comment:
```
ip := net.ParseIP(...)
if ip == nil {
  // probably should return an error here
}

if ip.To4() != nil {
} else {
}

if nameServer[0] == '[' && nameServer[len(nameServer) -1] == ']' {
  // you need to validate the string inside is valid ipv6?
}
```

Reply:
@bowei : We can't return an error when net.ParseIP(...) returns nil, since we need to fall through to 'err := validateHostAndPort(nameServer)'. The logic here is that if it's not a valid (stand-alone, without a port) IP addr, then we need to assume it's a host + port string, and validate it with validateHostAndPort() (and that function will vet the string and return an error if it's abnormal).
Re. "you need to validate the string inside is valid ipv6?"... that is a very good point, but looking at this now, this check for an IPv6 address inside brackets should not be here. Brackets should not be seen around "stand-alone" IPv6 address... we should only see brackets around IPv6 addresses when it appears as part of an IP:port format, or it appears as part of a URL. I'll delete this check for IPv6 inside brackets.